src/DIRAC/WorkloadManagementSystem/Agent/ScoutingJobStatusAgent.py

```diff
@@ -0,0 +1,229 @@
+""" Agent for scout job framework to monitor scout status
+    and update main job status according to scout status.
+"""
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.WorkloadManagementSystem.Client.JobStateUpdateClient import JobStateUpdateClient
+from DIRAC.WorkloadManagementSystem.DB.JobLoggingDB import JobLoggingDB
+
+
+class ScoutingJobStatusAgent(AgentModule):
+    """
+        This agent checks for jobs with Scouting status 
+        and manages the job status in relation to associated scout jobs
+    """
+
+    def __init__(self, *args, **kwargs):
+        """ c'tor
+        """
+        AgentModule.__init__(self, *args, **kwargs)
+
+        self.jobDB = None
+        self.logDB = None
+
+  #############################################################################
+    def initialize(self):
+        """Sets defaults
+        """
+
+        self.jobDB = JobStateUpdateClient()
```

Comment:
Maybe it was me not being clear, but this can't be a 1-to-1 substitution between `JobDB` and `JobStateUpdateClient`, as the 2 don't expose the same methods. For example, `JobStateUpdateClient` does not expose the `selectJobs` method used below.

Reply:
Ah I see. Does JobStateUpdateClient access only the methods found in [JobStateUpdateHandler](https://dirac.readthedocs.io/en/latest/CodeDocumentation/WorkloadManagementSystem/Service/JobStateUpdateHandler.html), or are there others?

Reply:
Only those.

Reply:
In general, I would do this, for added clarity:

```suggestion
        self.jobStateUpdate = JobStateUpdateClient()
```
then you'll need at least also

```python
        self.jobMonitoring = JobMonitoringClient()
```